### PR TITLE
gh-actions: test-on-iotlab: additionally report as XML

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Now that https://github.com/RIOT-OS/RIOT/pull/14851 we can activate this feature for the weekly tests.

### Testing procedure
I'll trigger a run. It then should contain a report.xml in each artifact directory.

### Issues/PRs references
Follow-up on https://github.com/RIOT-OS/RIOT/pull/14851.

### CODE REVIEW

**Comment by miri64** [comment]
> I'll trigger a run. It then should contain a report.xml in each artifact directory.

See https://github.com/miri64/RIOT/actions/runs/227160094

**Comment by miri64** [comment]
Forgot to install the dependencies ^^"

File: .github/workflows/test-on-iotlab.yml [hunk 1]
```diff
@@ -110,9 +110,13 @@ jobs:
             -l 1,site=${{ matrix.boards.iotlab.site }}+archi=${{ matrix.boards.iotlab.archi }})
           iotlab-experiment wait -i ${IOTLAB_EXP_ID}
           echo "::set-env name=IOTLAB_EXP_ID::${IOTLAB_EXP_ID}"
+      - name: Install Python dependencies
```
Comment (aabadie) [inline .github/workflows/test-on-iotlab.yml:113]: There is already a step that installs the Python dependencies: https://github.com/RIOT-OS/RIOT/pull/14871/files#diff-489558d2c142b44b698f71446b44a34cR84. Just add `junit-xml` to the existing list.

Comment (miri64) [reply .github/workflows/test-on-iotlab.yml:113]: Oops. Will do!

Comment (miri64) [reply .github/workflows/test-on-iotlab.yml]: Fixed and started another test run at https://github.com/miri64/RIOT/actions/runs/228647748

**Comment by aabadie** [comment]
Just curious, what is your plan with the generated xml file ?

**Comment by miri64** [comment]
> Just curious, what is your plan with the generated xml file ?

@MrKevinWeiss and a student of his are planning to consolidate all testing data in one place, as far as I understand. Using a unified format for this is already a good start for that. HiL tests, release tests, and with this test-on-iotlab use now all support jUnit XML as a basis for that. AFAIK @kaspar030 is also planning to provide jUnit XML support for Murdock.

**Review by aabadie — APPROVED** [review]
ACK

Please squash!

**Comment by miri64** [comment]
Squashed. But needs rebase due to the `check-label` action ^^"

**Comment by miri64** [comment]
Done

**Comment by aabadie** [comment]
The "check-label" still didn't triggered.

**Comment by miri64** [comment]
Yeah I noticed... Not sure why that is Oo.

**Comment by miri64** [comment]
Ah, since the it only looks out for labels, reviews and if a PR is opened or closed, it did not pick up the change ;-).

**Pull request merged**